### ISSUE TITLE
Invalid OpenAPI spec in aws_api_gateway_rest_api leads to plugin crash during destroy state refresh

### ISSUE THREAD

**Issue by myniva** [hashicorp/terraform-provider-aws#21181]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v1.0.8
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v3.61.0

### Affected Resource(s)

* aws_api_gateway_rest_api
* aws_api_gateway_stage
* aws_api_gateway_deployment

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "aws_api_gateway_rest_api" "this" {
  name = "Test API"
  body = "invalid OpenAPI spec"
}

resource "aws_api_gateway_deployment" "default" {
  rest_api_id = aws_api_gateway_rest_api.this.id

  triggers = {
    redeployment = sha1("invalid OpenAPI spec")
  }

  lifecycle {
    create_before_destroy = true
  }
}

resource "aws_api_gateway_stage" "default" {
  deployment_id = aws_api_gateway_deployment.default.id
  rest_api_id   = aws_api_gateway_rest_api.this.id
  stage_name    = "default"
}
```

### Debug Output

https://gist.github.com/myniva/351d1ed28a465316aa15d6fc0f4f854d

### Expected Behavior

`terraform destroy` should run successfully.

### Actual Behavior

The Terraform AWS provider plugin crashes, see logs in https://gist.github.com/myniva/351d1ed28a465316aa15d6fc0f4f854d

### Steps to Reproduce

1. Run `terraform apply` in a clean workspace. This will result in the following error: `Error: error creating API Gateway specification: BadRequestException: Invalid OpenAPI input.` Note: The REST API resource was created nevertheless and can also be found in the state file.
2. Run `terraform apply` again. This will fail again with the same error. Now, you will see two instances of a REST API and both of them can be found in the state file.
3. Run `terraform destroy`

### Important Factoids

n/a

### References

@ewbankkit IMO this bug is related to the following issues:

* #21079 
* #21019 
* #20811 
* #21132
* and many more...

**Comment by justinretzolk** [comment]
Hey @myniva 👋 Thank you for the update. Unfortunately, I think that may be something that isn't possible to get around given the bug that was previously there. If you manually correct the dangling resource (I'm assuming you may have already done that given the elapsed time), do any further issues seem to persist?

**Comment by myniva** [comment]
Hey @justinretzolk! I am not blocked by the original issue. But as written in my previous comment: If I do the exact same steps as described in the initial bug report, I still get an error during destroy. Afterwards, I have to manually cleanup on of the resources. So from my perspective, there is still something going wrong here.
